packages/api/amplify_api/lib/src/api_plugin_impl.dart

```diff
@@ -98,8 +113,12 @@ class AmplifyAPIDart extends AmplifyAPI {
       type: EndpointType.graphQL,
       apiName: apiName,
     );
-    return _clientPool[endpoint.name] ??= AmplifyAuthorizationRestClient(
-        endpointConfig: endpoint.config, baseClient: _baseHttpClient);
+    return _clientPool[endpoint.name] ??= AmplifyHttpClient(
```

Comment:
not really what this PR is about, but figured we want to use this client to have expected user agent content